python/api/component/serializers.py

```diff
@@ -0,0 +1,79 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=redefined-builtin
+
+from rest_framework import serializers
+from rest_framework.reverse import reverse
+
+from api.api_views import filter_actions, CommonAPIURL
+from api.action.serializers import ActionShort
+
+from cm import issue
+from cm import status_api
+from cm.models import Action
+
+
+class ComponentObjectUrlField(serializers.HyperlinkedIdentityField):
+    def get_url(self, obj, view_name, request, format):
+        kwargs = {'component_id': obj.id}
+        return reverse(view_name, kwargs=kwargs, request=request, format=format)
+
+
+class ComponentSerializer(serializers.Serializer):
+    id = serializers.IntegerField(read_only=True)
+    cluster_id = serializers.IntegerField(required=True)
```

Comment:
Please add the `service_id` field. Also, please bring the data for all endpoints to one view. `/api/v1/cluster/1/service/1/component/`, `/api/v1/service/1/component/` and `/api/v1/component/`, and also for detailed display.

Reply:
all fixed